slides/aml-10-021918-trees-forests/index.html

```diff
@@ -0,0 +1,724 @@
+tree_dot = export_graphviz(tree, out_file=None, feature_names=cancer.feature_names)
+print(tree_dot)```
+]
+
+???
+---
+# Visualizing trees with sklearn
+.center[
+![:scale 80%](images/img_11.png)
+]
+???
+---
+
+
+# Showing dot files in Jupyter
+ - First install graphvis C library:
+   conda install graphviz
+ - Then install graphviz python library:
+ - pip install graphviz
+
+ .center[
+ ![:scale 60%](images/img_12.png)
+ ]
+
+
+???
+---
+class:spacious
+
+# Parameter Tuning
+
+ - Pre-pruning and post-pruning (not in sklearn yet)
+ - Limit tree size (pick one):
+   max_depth
+   --
+   max_leaf_nodes
+   --
+   min_samples_split
+   --
+   (and more)
+???
+---
+class:spacious
+
+# No pruning
+.center[
+![:scale 100%](images/img_13.png)
+]
+???
+
+---
+
+# max_depth = 4
+
+.center[
+![:scale 100%](images/img_14.png)
+]
+
+???
+---
+
+# max_leaf_nodes = 8
+.center[
+![:scale 50%](images/img_15.png)
+]
+
+???
+---
+
+# min_samples_split = 50
+.center[
+![:scale 70%](images/img_16.png)
+]
+
+???
+---
+
+.smaller[
+```python
+from sklearn.model_selection import GridSearchCV
+param_grid = {'max_depth':range(1, 7)}
+grid = GridSearchCV(DecisionTreeClassifier(random_state=0),param_grid=param_grid,
+                    cv=10)
+grid.fit(X_train, y_train)
+```
+]
+.center[
+![:scale 70%](images/img_17.png)
+]
+
+???
+---
+.smaller[
+```python
+from sklearn.model_selection import GridSearchCV
+param_grid = {'max_leaf_nodes':range(2, 20)}
+grid = GridSearchCV(DecisionTreeClassifier(random_state=0),
+                    param_grid=param_grid, cv=10)
+grid.fit(X_train, y_train)
+```
+]
+.center[
+![:scale 70%](images/img_18.png)
+]
+
+???
+---
+.center[
+![:scale 95%](images/img_19.png)
+]
+
+???
+---
+.center[
+![:scale 95%](images/img_20.png)
+]
+???
+---
+#Extrapolation
+.center[
+![:scale 80%](images/img_33.png)
+]
+???
+---
+
+class: split-40
+# Instability
+
+.column[
+.tiny-code[
+```python
+iris = load_iris()
+X_train, X_test, y_train, y_test = train_test_split(iris.data,
+                                                    iris.target,
+                                                    stratify=iris.
+                                                    target,
+                                                    random_state=0)
+tree = DecisionTreeClassifier(max_leaf_nodes=6).fit(X_train,
+                                                   y_train)
+tree_dot = export_graphviz(tree,out_file=None,feature_names=
+                          iris.feature_names)
+graphviz.Source(tree_dot)
+```
+]
+![:scale 70%](images/img_21.png)
+]
+.column[
+.tiny-code[```python
+X_train, X_test, y_train, y_test = 
+train_test_split(iris.data, iris.target, stratify=iris.target, random
+                _state=1)
+tree = DecisionTreeClassifier(max_leaf_nodes=6).fit(X_train,y_train)
+tree_dot = export_graphviz(tree,out_file=None,feature_names=iris.
+                          feature_names)
+graphviz.Source(tree_dot)
+```
+]
+.center[![:scale 35%](images/img_22.png)
```

Comment:
you can see that the last of type column didn't work here: the scale of the image is much different from the scale of the other image because the column is not restricted. I'm not sure why you didn't use left-column and right-column.